android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -160,15 +174,36 @@ public void setValue(long value, boolean suppressEvent)
 	@Override
 	public void onTimeChanged(TimePicker view, int hourOfDay, int minute)
 	{
+		detectTimeChange();
+	}
+	
+	private void detectTimeChange() {
 		Calendar calendar = Calendar.getInstance();
-		calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
-		calendar.set(Calendar.MINUTE, minute);
-		if (!suppressChangeEvent) {
+		TimePicker picker = (TimePicker) getNativeView();
+		boolean hasChanged;
+		if (Build.VERSION.SDK_INT >= 23) {
+			hasChanged = lastSelectedHour != picker.getHour() || lastSelectMin != picker.getMinute();
```

Comment:
This will cause an "opcode" warning in the log when ran on an API Level 22 or lower device. Please change it to the following...

```java
private void detectTimeChange()
{
	TimePicker picker = (TimePicker) getNativeView();
	if (picker == null) {
		return;
	}

	int currentSelectedHour;
	if (Build.VERSION.SDK_INT >= 23) {
		currentSelectedHour = ApiLevel23.getHourFrom(picker);
	} else {
		currentSelectedHour = picker.getCurrentHour();
	}
	int currentSelectedMinute;
	if (Build.VERSION.SDK_INT >= 23) {
		currentSelectedMinute = ApiLevel23.getMinuteFrom(picker);
	} else {
		currentSelectedMinute = picker.getCurrentMinute();
	}

	// ...
}

private static class ApiLevel23
{
	private ApiLevel23() {}

	public static int getHourFrom(TimePicker picker)
	{
		return picker.getHour();
	}

	public static int getMinuteFrom(TimePicker picker)
	{
		return picker.getMinute();
	}
}
```